bgpv2: Refactor service route policy rendering logic

rastislavs opened the pull request:
Use common method for rendering service route policies for all types of svc advertisements instead of 3 different methods, to avoid code repetition. Rely on exiting path rendering methods (`get<svcType>IPPaths`) instead of copying the same logic in the policy rendering methods.

Also move prefix length rendering to `get<svcType>IPPaths` and properly zero masked-out bits (e.g. to produce `192.168.1.0/24` instead of `192.168.1.1/24`) to produce proper keys for the `PathMap`.

rastislavs committed:
bgpv2: Refactor service route policy rendering logic

Use common method for rendering service route policies
for all types of svc advertisements instead of 3 different
methods, to avoid code repetition. Rely on exiting path
rendering methods (get<svcType>IPPaths) instead of copying
the same logic in the policy rendering methods.

Also move prefix length rendering to get<svcType>IPPaths and
properly zero masked-out bits (e.g. to produce 192.168.1.0/24
instead of 192.168.1.1/24) to produce proper keys for the PathMap.

Signed-off-by: Rastislav Szabo <rastislav.szabo@isovalent.com>

harsimran-pabla commented on pkg/bgpv1/manager/reconcilerv2/service.go:
```diff
@@ -871,21 +735,19 @@ func serviceLabelSet(svc *slim_corev1.Service) labels.Labels {
 
-func getPrefixes(family types.Family, svc *slim_corev1.Service, advert v2.BGPAdvertisement, addr netip.Addr, v4Prefixes, v6Prefixes types.PolicyPrefixMatchList) (types.PolicyPrefixMatchList, types.PolicyPrefixMatchList) {
-	mask := addr.BitLen()
+func getServicePrefixLength(svc *slim_corev1.Service, addr netip.Addr, advert v2.BGPAdvertisement) int {
```
Just noticed, do we support cluster IP route aggregation ? In which case, it should consider iTP instead of eTP. 

It does not need to be part of this change, maybe a follow up to address it.

rastislavs replied:
Good point, fixing it in a separate commit in this PR, as it is related and this PR needs to wait until release anyway.

harsimran-pabla reviewed (approved):
Thanks, nice cleanup 👍

rastislavs committed:
bgpv2: Properly handle aggregation prefix length for ClusterIP services

For ClusterIP service advertisements we need to consider
internalTrafficPolicy instead of externalTrafficPolicy.

Signed-off-by: Rastislav Szabo <rastislav.szabo@isovalent.com>
